training/components.py

```diff
@@ -167,111 +169,44 @@ def list_phase1_final_model():
-                              --save_samples={save_samples} \
-                              --log_level=INFO \
-                              --max_batch_len={max_batch_len} \
-                              --seed={seed} \
-                              --cpu_offload_optimizer \
-                              --cpu_offload_params \
-                              --distributed_training_framework fsdp \
-                              --is_granite \
-                              --checkpoint_at_epoch
-                      command:
-                        - /bin/bash
-                        - '-c'
-                        - '--'
-                      image: {image}
-                      name: pytorch
-                      volumeMounts:
-                        - mountPath: /input_data
-                          name: input-data
-                          readOnly: true
-                        - mountPath: /input_model
-                          name: model
-                          readOnly: true
-                        - mountPath: /output
-                          name: output
-                      env:
-                        - name: NNODES
-                          value: \"{nnodes}\"
-                        - name: NPROC_PER_NODE
-                          value: \"{nproc_per_node}\"
-                        - name: XDG_CACHE_HOME
-                          value: /tmp
-                        - name: TRITON_CACHE_DIR
-                          value: /tmp
-                        - name: HF_HOME
-                          value: /tmp
-                        - name: TRANSFORMERS_CACHE
-                          value: /tmp
-                      resources:
-                        requests:
-                          cpu: 8
-                          "nvidia.com/gpu": {nproc_per_node}
-                        limits:
-                          cpu: 8
-                          "nvidia.com/gpu": {nproc_per_node}
-                  volumes:
-                    - name: input-data
-                      persistentVolumeClaim:
-                        claimName: {input_pvc_name}
-                    - name: model
-                      persistentVolumeClaim:
-                        claimName: {model_pvc_name}
-                    - name: output
-                      persistentVolumeClaim:
-                        claimName: {output_pvc_name}
-            Worker:
-              replicas: {nnodes-1}
-              restartPolicy: OnFailure
-              template:
-                metadata:
-                  annotations:
-                    sidecar.istio.io/inject: 'false'
-                spec:
-                  containers:
-                    - args:
-                        - |
-                          echo "Running phase {phase_num}"
+    resources_per_worker = {"cpu": "8", "nvidia.com/gpu": nproc_per_node}
+
+    base_image = "quay.io/redhat-et/ilab:shrey"
```

Comment:
Should this images be `RHELAI_IMAGE` and not this custom image `quay.io/redhat-et/ilab:shrey`?

Reply:
Yes, the value here should be the same as `RHELAI_IMAGE`, but due to KFP limitations, the Pythonic components are hermetic and you can't reference external constanstants here...

https://www.kubeflow.org/docs/components/pipelines/user-guides/components/lightweight-python-components/#python-function-requirements

I think we can get around that if we pass the image for `PytorchJob` as component parameter which defaults to the `RHELAI_IMAGE` constant.